sdk/src/wallet/account/operations/transaction/prepare_output.rs

```diff
@@ -97,263 +98,194 @@ where
             }
         }
 
-        let first_output = first_output_builder.finish(token_supply)?;
+        // Build output with minimum required storage deposit so we can use the amount in the next step
+        let first_output = first_output_builder
+            .with_minimum_storage_deposit(rent_structure)
```

Comment:
this minimum storage deposit stuff feels really tacked on, and i dont mean this one specifically, but the entire builder,
not sure how to improve it easily though

Reply:
Yeah in prepare_output it's very complex because new features/unlock conditions change it and with the available base coin amount you have basically 3 values which can affect the others and if you change one you have to check if it's still fine with the others
Not sure what we can do better there